mythril/laser/ethereum/svm.py

```diff
@@ -70,13 +76,33 @@ def __init__(
 
         self.time = None
 
-        self.pre_hooks = {}
-        self.post_hooks = {}
+        self.pre_hooks = defaultdict(list)
+        self.post_hooks = defaultdict(list)
+
+        self.register_detection_modules()
 
         logging.info(
             "LASER EVM initialized with dynamic loader: " + str(dynamic_loader)
         )
 
+    def register_detection_modules(self):
```

Comment:
I would like to keep the logic that doesn't have to do with symbolic execition but just reasons about the execution, outside of laser. WDYT

Reply:
So what I'd like to happen is that the equivalent class of the sym exec wrapper, registers the hooks with the laser evm instance. But that the laser does not get dependencies towards mythril.analysis

Reply:
I'll refactor it into a utility method :+1:

Reply:
To elaborate a little more: I think it's reasonable to keep the modules that reason about execution outside of Laser. That would be a semantic separation, which is one way of looking at it (and the better one for us regarding potential future refactorings). The other one would be the view that the detection modules change the state of Laser and (to satisfy the isolation property) should be included.

I'll add the `register_post_hook` and `register_pre_hook` methods to Laser that will allow you to register any of them. That way we have the same functionality without the analysis module dependencies.